src/ForqueSensorHW.cpp

```diff
@@ -0,0 +1,91 @@
+
+#include "forque_sensor_hardware/ForqueSensorHW.h"
+
+namespace forqueSensorHW {
+
+//==============================================================================
+ForqueSensorHW::ForqueSensorHW(std::string sensorName, std::string frameId,
+                               std::string ipAddress)
+    : sensorName(sensorName), frameId(frameId), address(ipAddress),
+      biasState(pr_hardware_interfaces::TRIGGER_IDLE) {
+
+  force.fill(0);
+  torque.fill(0);
+  bias.fill(0);
+
+  // Register Handles
+  hardware_interface::ForceTorqueSensorHandle forqueSensorHandle(
+      sensorName, frameId, &force[0], &torque[0]);
+  forceTorqueInterface.registerHandle(forqueSensorHandle);
+  hardware_interface::InterfaceManager::registerInterface(
+      &forceTorqueInterface);
+
+  pr_hardware_interfaces::TriggerableHandle biasHandle("/biasTrigger",
+                                                       &biasState);
+  biasTriggerInterface.registerHandle(biasHandle);
+  hardware_interface::InterfaceManager::registerInterface(
+      &biasTriggerInterface);
+}
+
+//==============================================================================
+bool ForqueSensorHW::connect() {
+  try {
+    netft = std::unique_ptr<netft_rdt_driver::NetFTRDTDriver>(
+        new netft_rdt_driver::NetFTRDTDriver(address));
+  } catch (std::runtime_error e) {
+    netft = nullptr;
+    std::cerr << "Error when starting NetFT device: " << e.what() << std::endl;
+    return false;
+  }
+  return true;
+}
+
+//==============================================================================
+void ForqueSensorHW::update() {
+
+  if (biasState == pr_hardware_interfaces::TRIGGER_REQUESTED) {
+    shouldCollectBiasData = true;
```

Comment:
I am confused why `bias` is collected when `TRIGGER` is requested. Could you explain?

Reply:
MoveUntilTouchController uses trigger to calibrate bias